src/crewai/knowledge/storage/knowledge_storage.py

```diff
@@ -35,9 +35,16 @@ class KnowledgeStorage(BaseKnowledgeStorage):
     """
 
     collection: Optional[chromadb.Collection] = None
+    store_dir: Optional[str] = "knowledge"
+    app: Optional[chromadb.PersistentClient] = None
 
-    def __init__(self, embedder_config: Optional[Dict[str, Any]] = None):
-        self._initialize_app(embedder_config or {})
+    def __init__(
+        self,
+        embedder_config: Optional[Dict[str, Any]] = None,
+        store_dir: Optional[str] = None,
+    ):
+        self.embedder_config = embedder_config
+        self.store_dir = store_dir
```

Comment:
Won't this always overwrite store_dir that we want to default to knowledge?

Reply:
its more collection_name that it is store_dir. i need to change this. it points to the same directory it persists in just different collection